source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -0,0 +1,283 @@
+matches your query filter. 
+
+The positional ``$`` operator references the array matched by your query
+filter and cannot reference an embedded array. If you want to access an
+embedded array, use the :ref:`filtered positional operator <filtered_positional_operator>`. 
+
+Example
+```````
+
+The following example performs the following actions:
+
+- Query for embedded arrays where "B" exists in warehouses
+- Use the ``inc()`` method of the ``Updates`` builder to decrement the ``qty`` field by "3" 
+- Use a positional ``$`` operator to indicate an update to the first matched embedded array
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin updateValueExample
+   :end-before: end updateValueExample
+
+The output of the above code should look like this:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 1, 
+      "color": "green", 
+      "instock": [
+        { "qty": 8, "warehouse": ["A", "E"] }, 
+        { "qty": 10, "warehouse": ["B", "C"] }, 
+        { "qty": 15, "warehouse": ["B", "F"] } ]}
+
+For additional information on the operator and method mentioned in this
+section, see the following resources: 
+
+- :manual:`Positional $ Operator </reference/operator/update/positional>` Server Manual Explanation
+- :java-core-api:`inc() <com/mongodb/client/model/Updates.html#inc(java.lang.String,java.lang.Number)>` API Documentation
+
+All Array Elements
+~~~~~~~~~~~~~~~~~~
+
+Use the all positional ``$[]`` operator to update each embedded array in
+the document. 
+
+Example
+```````
+
+The following example performs the following actions: 
+
+- Using the ``inc()`` method of the ``Updates`` builder to increment the ``qty`` field by "5"
+- Use an all positional operator to indicate an update to each embedded array
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin updateAllElementsExample
+   :end-before: end updateAllElementsExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 1, 
+      "color": "green", 
+      "instock": [
+        {"qty": 13, "warehouse": ["A", "E"]}, 
+        {"qty": 18, "warehouse": ["B", "C"]}, 
+        {"qty": 20, "warehouse": ["B", "F"]} ]}
+
+For additional information on the operator and method mentioned in this
+section, see the following resources: 
+
+- :manual:`Positional $[] Operator </reference/operator/update/positional-all/>` Server Manual Explanation
+- :java-core-api:`inc() <com/mongodb/client/model/Updates.html#inc(java.lang.String,java.lang.Number)>` API Documentation
+
+.. _filtered_positional_operator:
+
+Filtered Array Elements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify a ``UpdateOptions.arrayFilters`` condition and use the filtered
+positional ``$[<identifier>]`` operator to update certain embedded
+arrays in a document. 
```

Comment:
```suggestion
To match array elements using a filter, use the filtered
positional ``$[<identifier>]`` operator. You must specify a
list of array filters in your update operation to use a filtered
positional ``$[<identifier>]`` operator.
```

Issue: I believe this section should start with the task it teachers readers how to perform (task oriented docs). 

Suggestion: I believe the "specifying an array filter" statement could work well as a prerequisite "you must" for using a filtered positional operator.

Reply:
Updated to:
```suggestion
You must include an array filter to specify which array elements to update.
```